src/plot.js

```diff
@@ -148,6 +173,24 @@ function defaultFilter(index, channels, values) {
   return index;
 }
 
+function addScaleChannels(scaleChannels, channels, options) {
+  for (const [, channel] of channels) {
+    const {scale} = channel;
+    if (scale !== undefined) {
+      const {percent, transform = percent ? x => x * 100 : undefined} = options[scale] || {};
+      if (transform != null) channel.value = Array.from(channel.value, transform);
+      if (scaleChannels.has(scale)) scaleChannels.get(scale).push(channel);
+      else scaleChannels.set(scale, [channel]);
+    }
+  }
+}
+
+function applyLayout(layout, index, scaleDescriptors, channels, dimensions, options) {
+  const scales = ScaleFunctions(scaleDescriptors);
+  const values = channels ? applyScales(channels, scales) : {};
+  return layout ? layout.call(this, index, scaleDescriptors, values, dimensions, options) : {index, values};
```

Comment:
This is passing the entire *options* object to *mark*.layout. I don’t think we want to expose that much information to the layout—it risks making the layouts unwieldy because their behavior could be dependent on anything in the plot specification. It looks like this *options* is only needed for faceting (not by our current hexbin and dodge layouts), so, I suspect we’ll need to find a different way to implement faceted layouts to avoid exposing *options*.

Reply:
This is also passing *scaleDescriptors* to *mark*.layout, but I think we want *scales* (i.e., scale functions) for symmetry with *mark*.render. In any case the scales are not used by the current layouts.